File: app/src/main/java/com/example/pharmasync/AppDatabase.java
package com.example.pharmasync;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {MedicineEntity.class, SupplierEntity.class, OrderEntity.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {
    public abstract MedicineDao medicineDao();
    public abstract SupplierDao supplierDao();
    public abstract OrderDao orderDao();

    private static volatile AppDatabase INSTANCE;

    public static AppDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (AppDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(
                            context.getApplicationContext(),
                            AppDatabase.class,
                            "pharmasync_room.db"
                    ).allowMainThreadQueries().fallbackToDestructiveMigration().build();
                }
            }
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/example/pharmasync/data/local/AppDatabase.kt
package com.example.pharmasync.data.local

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

/**
 * Local copy of the signed-in user's data. Firestore is the source of truth; snapshot
 * listeners mirror it here so screens render instantly and keep working offline.
 */
@Database(
    entities = [MedicineEntity::class, OrderEntity::class, SupplierEntity::class, InvoiceEntity::class],
    version = 2,
    exportSchema = false,
)
abstract class AppDatabase : RoomDatabase() {
    abstract fun medicineDao(): MedicineDao
    abstract fun orderDao(): OrderDao
    abstract fun supplierDao(): SupplierDao
    abstract fun invoiceDao(): InvoiceDao

    companion object {
        fun create(context: Context): AppDatabase =
            Room.databaseBuilder(context.applicationContext, AppDatabase::class.java, "pharmasync.db")
                // The database is a cache of Firestore, so it is safe to rebuild on schema changes.
                .fallbackToDestructiveMigration()
                .build()
    }
}
